pages/contact.tsx

```diff
@@ -57,86 +60,88 @@ const Contact: NextPage = () => {
     company: '',
     comments: '',
   });
-  let isValid = true;
+
+  let errors = { ...validation };
   const checkValidation = () => {
-    let errors = validation;
-    console.log('errors', errors);
+    console.log('errors', validation);
 
-    //full Name validation
-    if (!inputValues.fullName.trim()) {
-      if (errors.fullName) {
-        errors.fullName = 'First name is required';
-        isValid = false;
-      }
-    } else if (inputValues.fullName.length < 5) {
-      errors.fullName = 'First name not less than 5 char';
-      isValid = false;
-    } else if (inputValues.fullName.length > 10) {
-      errors.fullName = 'First name not more than 10 char';
-      isValid = false;
+    if (!inputValues.fullName) {
+      errors.fullName = 'First name is required';
     } else {
       errors.fullName = '';
-      isValid = true;
     }
+
     // email validation
-    const emailCond =
-      "/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:.[a-zA-Z0-9-]+)*$/";
-    if (!inputValues.email.trim()) {
+    // const emailCond = "/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:.[a-zA-Z0-9-]+)*$/";
+    if (!inputValues.email) {
       errors.email = 'Email is required';
-      isValid = false;
-    } else if (inputValues.email.match(emailCond)) {
+    } else if (
+      !inputValues.email
+        .toLowerCase()
+        .match(
+          /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i,
+        )
+    ) {
       errors.email = 'Please ingress a valid email address';
-      isValid = false;
     } else {
       errors.email = '';
-      isValid = true;
     }
 
-    if (!inputValues.phoneNumber.trim()) {
+    if (!inputValues.phoneNumber) {
       errors.phoneNumber = 'phoneNumber is required';
-      isValid = false;
-    } else if (inputValues.phoneNumber.length < 9) {
-      errors.phoneNumber = 'phoneNumber must be correct';
-      isValid = false;
-    } else if (inputValues.phoneNumber.length >= 11) {
-      isValid = false;
-      errors.phoneNumber = 'number must be valid';
+    } else if (inputValues.phoneNumber.length > 10) {
+      errors.phoneNumber = 'phoneNumber not exceed 10';
+    } else if (inputValues.phoneNumber.length < 10) {
+      errors.phoneNumber = 'phoneNumber not less than 10';
     } else {
       errors.phoneNumber = '';
-      isValid = true;
     }
 
-    if (!inputValues.company.trim()) {
+    if (!inputValues.company) {
       errors.company = 'company name is required';
-      isValid = false;
-    } else if (inputValues.company.length < 5) {
-      errors.company = 'company name must be filled';
-      isValid = false;
-    } else if (inputValues.company.length > 10) {
-      errors.company = 'First name not more than 10 char';
-      isValid = false;
     } else {
       errors.company = '';
-      isValid = true;
     }
     return setValidation(errors);
   };
 
   useEffect(() => {
     checkValidation();
```

Comment:
Instead of doing this you can check them directly inside the change handler, separately for each field